src/transformStateWithClones.js

```diff
@@ -6,8 +7,62 @@
  *
  * @return {Object[]}
  */
+// function transformStateWithClones(state, actions) {
+//   // write code here
+//   const table = {
+//     ...state,
+//   };
+//   const result = [];
+
+//   for (const action of actions) {
+//     if (action.type === 'clear') {
+//       for (const delet in table) {
+//         delete table[delet];
+//       }
+//       console.log(table);
+//     } else if (action.type === 'removeProperties') {
+//       for (const remove of action.keysToRemove) {
+//         delete table[remove];
+//       }
+//     } else {
+//       Object.assign(table, action.extraData);
+//     }
+
+//     result.push({ ...table });
+//   }
+
+//   return result;
+// }
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyOfState = {
+    ...state,
```

Comment:
The copy of the state is created only once before the loop. This will lead to all actions mutating the same state copy, which violates the task requirement to not mutate the initial state object and to return an array with states received after each action. You should create a new copy of the state within the loop before applying each action. This can be achieved by moving the copy creation inside the loop, ensuring that each action works with its own separate copy.